packages/playground/src/components/networks.vue

```diff
@@ -1,154 +1,109 @@
+          />
+
+          <NetworkItem
+            :tooltip-text="wireguardTooltip"
+            label="Add Wireguard Access"
+            :value="$props.wireguard"
+            :emit-function="readonlyWireguard ? undefined : $attrs['onUpdate:wireguard']"
+          />
+
+          <v-alert
+            v-if="error"
+            class="my-2"
+            type="warning"
+            variant="tonal"
+            text="You must enable at least one of network options."
+          />
+        </v-expansion-panel-text>
+      </v-expansion-panel>
+    </v-expansion-panels>
+  </input-tooltip>
 </template>
 
 <script lang="ts">
 import { noop } from "lodash";
-import { computed, getCurrentInstance, onMounted, onUnmounted, type PropType, ref, watch } from "vue";
+import { computed, getCurrentInstance, onMounted, onUnmounted, ref, watch } from "vue";
 
 import { useForm, ValidatorStatus } from "@/hooks/form_validator";
 import type { InputValidatorService } from "@/hooks/input_validator";
-import type { DomainInfo } from "@/types/nodeSelector";
+
+import NetworkItem from "./NetworkItem.vue";
+
+interface CreateNetworksOptions {
+  ipv4: boolean;
+  ipv6: boolean;
+  planetary: boolean;
+  mycelium: boolean;
+  wireguard: boolean;
+}
+
+export function useNetworks(options: Partial<CreateNetworksOptions> = {}) {
+  return {
+    ipv4: ref(options.ipv4 ?? false),
+    ipv6: ref(options.ipv6 ?? false),
+    planetary: ref(options.planetary ?? false),
+    mycelium: ref(options.mycelium ?? true),
+    wireguard: ref(options.wireguard ?? false),
+  };
+}
 
 export default {
   name: "Network",
+  components: { NetworkItem },
   props: {
-    required: {
-      type: Boolean,
-      default: () => false,
-    },
-    ipv4: {
-      type: Boolean,
-      default: () => null,
-    },
-    ipv6: {
-      type: Boolean,
-      default: () => null,
-    },
-    planetary: {
-      type: Boolean,
-      default: () => null,
-    },
-    mycelium: {
-      type: Boolean,
-      default: () => null,
-    },
-    wireguard: {
-      type: Boolean,
-      default: () => null,
-    },
-    disabled: { type: Boolean },
-    enableIpv4: {
-      type: Boolean,
-      default: () => null,
-    },
-    domain: {
-      type: Object as PropType<DomainInfo>,
-      required: false,
-    },
-  },
-  emits: {
-    "update:ipv4": (value?: boolean) => value,
-    "update:ipv6": (value?: boolean) => value,
-    "update:planetary": (value?: boolean) => value,
-    "update:mycelium": (value?: boolean) => value,
-    "update:wireguard": (value?: boolean) => value,
+    required: { type: Boolean, default: () => false },
+    ipv4: { type: Boolean, default: () => null },
+    ipv6: { type: Boolean, default: () => null },
+    planetary: { type: Boolean, default: () => null },
+    mycelium: { type: Boolean, default: () => null },
+    wireguard: { type: Boolean, default: () => null },
+    hasCustomDomain: { type: Boolean, default: () => false },
```

Comment:
please note that the only case we have wg not `readOnly` is 
**having custom domain with ipv4 enalbled** 


[Screencast from 2024-09-01 14-57-56.webm](https://github.com/user-attachments/assets/4d4e0c5c-5f25-48e9-86e8-ddd5e17a086d)

Reply:
shouldn't we enable it if user didn't ask for custom domain at all?

Reply:
no if we have custom domain without ipv4 we have to create fqdn record on gw node, but if we have ipv4 and custom domain the user should map the machine ip to the custom domain 
please check https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3287#issuecomment-2301993915